documentation/content/content-nodes.html

```diff
@@ -213,94 +217,69 @@ <h3 id="master-election">Master election</h3>
-</p><p>
-To handle a cluster state version reset,
-clients have some functionality for detecting whether distributors seems to agree
-the cluster state it has is wrong, even if version is newer.
-In such cases clients will dump its cluster state and use a new one even if version is lower.
-Clients thus handle a version reset,
-but it may result in a few temporarily failures while it is confused.
-</p>
-
-
-
 <h2 id="distributor">Distributor</h2>
 <p>
-The distributor keeps track of which search nodes should hold which copy of each bucket,
-based on the <em>redundancy</em> setting and information from the cluster controller.
-Distributors are responsible for keeping track of metadata for a non-overlapping subset of the data in the cluster.
-The distributors each keep a bucket database containing metadata of buckets it is responsible for.
-This metadata indicates what content nodes store copies of the buckets,
-the checksum of the bucket content and the number of documents and meta entries within the bucket.
-Each incoming document is assigned to a bucket and forwarded to the right search nodes.
-All requests related to these buckets are sent through the distributors.
+Document operations are mapped to content nodes based on bucket locations -
+each put/update/get/remove is mapped to a <a href="buckets.html">bucket</a>
+and sent to the right content nodes.
+To manage the document set as it grows and nodes change, buckets move.
+I.e, sets of documents (a bucket) is copied between nodes for redundancy /
+redistribution for growth / node failures.
 </p><p>
-A <a href="../elastic-vespa.html#bucket-distribution-algorithm">distribution algorithm</a>
-calculates what distributors are responsible for given buckets based on
-information of what nodes are available in the cluster.
-Clients calculate what distributor to talk to using information in the cluster state.
+Buckets are mapped to distributors using the <a href="idealstate.html">ideal state algorithm</a>.
+As the cluster state changes, buckets are re-mapped immediately.
+The mapping does not overlap - a bucket is owned by one distributor.
 </p><p>
-When a distributor goes down or becomes available,
-all distributors reshuffle what buckets they are responsible for.
-During this reshuffling, clients may get some requests temporarily failing
-as distributors taking over bucket ownership doesn't yet know where copies of its buckets are located.
-These requests will automatically be resent, providing timeouts allow for it.
-Changes in what distributors are up and available will currently cause windows of availability loss.
-Other distributors take over bucket ownership.
-To do this, they fetch bucket metadata for new buckets from all storage nodes,
-in-memory operations for speed.
+Distributors do not persist the bucket database,
+the bucket-to-content-node mapping is stored in memory in the distributor.
+Document count per bucket is stored as well. <!-- ToDo: checksum? -->
```

Comment:
Currently document count, persisted size and a metadata checksum is stored per replica per bucket